### ISSUE TITLE
vue-cli vuex "refusing to install vuex as dependency of itself"

### ISSUE THREAD

**Issue by ChrisCates** [vuejs/vuex#133]
Any ideas? Just trying out VueX... Trying to learn this stuff...

**Comment by yyx990803** [comment]
You're most likely doing `npm install vuex` in the wrong directory.

**Comment by ChrisCates** [comment]
@yyx990803 Root directory of project is incorrect?
```
npm install vuex --save
```

**Comment by yyx990803** [comment]
This repo is the source code for vuex... you install vuex in other projects to use it.

**Comment by ChrisCates** [comment]
@yyx990803 I don't think you're understanding the context of the problem. The NPM install command throws:
```
refusing to install vuex as dependency of itself
```

**Comment by yyx990803** [comment]
I don't think you provided enough context. This would only happen if you are installing vuex inside a directory where vuex's `package.json` is present, so there must be something wrong with your project setup.

**Comment by ChrisCates** [comment]
@yyx990803 Thanks for the clarification. The project was actually generated by the `vue-cli` npm package. And is the standard webpack template.

**Comment by ChrisCates** [comment]
@yyx990803 OMG, I figured it out. I named the project "vuex". Even though it was not the vuex repository.

So sorry for wasting your time.

**Comment by blake-newman** [comment]
Haha I've done that before,  no worries it's a mistake anyone could make. Perhaps we can add a check in cli.  With a warning `Project name conflicts with a prexisting npm package,  this could cause undesired effects.`

**Comment by ruut1234** [comment]
open file package.json and change package name

**Comment by eldanizguluzade** [comment]
> open file package.json and change package name

Correct

**Comment by kennyendowed** [comment]
trying to install vuex in vue 3 but am getting the bellow error

npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: portfolio@0.1.0
npm ERR! Found: vue@3.1.1
npm ERR! node_modules/vue
npm ERR!   vue@"^3.0.0" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer vue@"^2.0.0" from vuex@3.6.2
npm ERR! node_modules/vuex
npm ERR!   vuex@"*" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency `resolution.`

**Comment by cuebit** [comment]
@kennyendowed `npm install vuex@next`

**Comment by kylecao413** [comment]
Trying on vue 2 project . I got this
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR!
npm ERR! While resolving: hemkraft@1.0.0
npm ERR! Found: vue@2.7.13
npm ERR! node_modules/vue
npm ERR!   vue@"^2.5.2" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer vue@"^3.0.2" from vuex@4.0.2
npm ERR! node_modules/vuex
npm ERR!   vuex@"4.0.2" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR!
npm ERR! See C:\Users\Kyle Cao\AppData\Local\npm-cache\eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\Kyle Cao\AppData\Local\npm-cache\_logs\2022-11-02T04_12_39_000Z-debug-0.log
